chainer/functions/normalization/batch_normalization.py

```diff
@@ -75,8 +75,15 @@ def forward(self, inputs):
             cudnn_mode = self.mode.get_cudnn_mode()
             libcudnn.deriveBNTensorDescriptor(derivedBnDesc.value,
                                               x_desc.value, cudnn_mode)
-            one = numpy.array(1, dtype=dtype).ctypes
-            zero = numpy.array(0, dtype=dtype).ctypes
+            dtype_param = _get_dtype_of_tensor_descriptor(derivedBnDesc)
+            if dtype_param is not dtype:
+                gamma = gamma.astype(dtype_param)
+                beta = beta.astype(dtype_param)
+                self.running_mean = self.running_mean.astype(dtype_param)
+                self.running_var = self.running_var.astype(dtype_param)
```

Comment:
It looks that the original running mean/var arrays are not updated by `batchNormalizationForwardTraining`. Could you add code that writes back the updated values to the original arrays?

Reply:
Ah, I did not notice that. That is a problem. Will fix codes so that the original arrays of running mean/var arrays are updated appropriately.